advanced/Templates/pihole-FTL.service

```diff
@@ -57,10 +59,15 @@ start() {
     fi
 
     if setcap CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN,CAP_SYS_NICE,CAP_IPC_LOCK,CAP_CHOWN+eip "/usr/bin/pihole-FTL"; then
-      su -s /bin/sh -c "/usr/bin/pihole-FTL" pihole || exit $?
+      su -s /bin/sh -c "/usr/bin/pihole-FTL" pihole || ec=$?
     else
       echo "Warning: Starting pihole-FTL as root because setting capabilities is not supported on this system"
-      /usr/bin/pihole-FTL || exit $?
+      /usr/bin/pihole-FTL || ec=$?
+    fi
+    # Cleanup if startup failed
+    if [ "${ec}" != 0 ]; then
+        cleanup
+        exit "${ec}"
```

Comment:
Since you're now trapping for non-zero exit you don't need any of this code. 
`/usr/bin/pihole-FTL` is all you need.